repositories.bzl

```diff
@@ -1,22 +1,295 @@
+# Copyright 2017 Istio Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+#
+load("@io_bazel_rules_go//go:def.bzl", "go_repositories", "new_go_repository", "go_repository")
+load("@org_pubref_rules_protobuf//gogo:rules.bzl", "gogo_proto_repositories")
 
-# use_local flag is used to control
-def new_git_or_local_repository(
-    name,
-    build_file,
-    path,
-    commit,
-    remote,
-    use_local = False):
+ISTIO_API_SHA = "a0ba5903ae2771eac13f80c3a936ed70fd3494d5"   # Aug 16, 2017 (no release)
+
+def go_istio_api_repositories(use_local=False):
```

Comment:
can we just make this a separate `api_repositories.bzl` that this script loads and calls?  that way it can be used elsewhere too.